troubadix/plugins/spelling.py

```diff
@@ -63,177 +60,193 @@ def run(self) -> Iterator[LinterResult]:
             codespell_ignore = f"{codespell_config_path}/codespell.ignore"
 
         # Run codespell as internal process
-        with redirect_stdout(io.StringIO()) as codespell:
-            codespell_main(
-                *(
-                    "--hard-encoding-detection",
-                    "--dictionary=-",
-                    f"--dictionary={codespell_additions}",
-                    f"--exclude-file={codespell_exclude}",
-                    f"--ignore-words={codespell_ignore}",
-                    "--disable-colors",
-                    f"{str(self.context.nasl_file)}",
-                )
+        batch_size = 10_000
+        for i in range(0, len(self.context.nasl_files), batch_size):
+            # files_parameter = [
+            #     str(nasl_file)
+            #     for nasl_file in self.context.nasl_files[i : i + batch_size]
+            # ]
+
+            files_parameters = tuple(
+                [
+                    str(nasl_file)
+                    for nasl_file in self.context.nasl_files[i : i + batch_size]
+                ]
             )
-
-        codespell = codespell.getvalue()
-        if "Traceback (most recent call last):" not in codespell:
-            _codespell = codespell.splitlines()
-            codespell = ""
-            for line in _codespell:
-
-                # From /Policy which is just a huge blob of text and too large
-                # for codespell.exclude:
-                if "policy_file_checksums_win.nasl" in line:
-                    if re.search(r"nD\s+==>\s+and, 2nd", line) or re.search(
-                        r"oD\s+==>\s+of", line
+            codespell_arguments = (
+                "--hard-encoding-detection",
+                "--dictionary=-",
+                f"--dictionary={codespell_additions}",
+                f"--exclude-file={codespell_exclude}",
+                f"--ignore-words={codespell_ignore}",
+                "--disable-colors",
+            ) + files_parameters
```

Comment:
IMHO you should use lists for concatenating the arguments

Reply:
And then just go for tuple creation right when required ?

Reply:
I suppose you did use a tuple because you tough this is required for the star operator. It's not :-)
```python
>>> a = ["a", "b"]
>>> def foo(*args):
...     print(args)
... 
>>> foo(*a)
('a', 'b')
```

Reply:
You'd be right about that